include/solvers/navier_stokes_base.h

```diff
@@ -370,7 +370,9 @@ class NavierStokesBase : public PhysicsSolver<VectorType>
    * Post-processing as parallel VTU files
    */
   void
-  write_output_results(const VectorType &solution);
+  write_output_results(
+    const VectorType &             solution,
+    std::shared_ptr<Function<dim>> additional_scalar = nullptr);
```

Comment:
I'm not sure that's a very clean design... In essence, if you wanted to add other type of functions or scalars or something from derived navier-stokes solver, you would always end-up modifying this interface.
The best solution would be to add a small virtual function that would be verysimilar to the one called for the multiphysics. If you did that, you could add any field you want without modifying the base interface and it would be coherent with what is done with the multiphysics interface. Again you can come and see me if you have questions. :)